src/ducks/categorization/services.js

```diff
@@ -186,6 +204,37 @@ const getLocalClassifierOptions = transactionsWithManualCat => {
   }
 }
 
+export const atomicReweight = (classifier, category, word, frequencyCount) => {
+  const newFrequencyCount = 1 + Math.log(frequencyCount)
+  const deltaFrequencyCount = frequencyCount - newFrequencyCount
+  // update the right entries of the classifier's parameters
+  classifier.vocabulary[word] -= deltaFrequencyCount
+  classifier.wordCount[category] -= deltaFrequencyCount
+  classifier.wordFrequencyCount[category][word] = newFrequencyCount
+}
+
+export const reweightModel = classifier => {
+  // loop over categories in the wordFrequencyCat attribute
+  const wordFrequencyCount = classifier.wordFrequencyCount
+  // for each category
+  for (const category in wordFrequencyCount) {
+    if (wordFrequencyCount.hasOwnProperty(category)) {
```

Comment:
`for (const category in Object.keys(wordFrequencyCount))` will help get rid of the `if` and the extra nesting level :)

BTW, to help with unnecessary nesting, an early continue is clearer.

```
for () {
  if (!condition) { continue }

  // rest of the loop is not indented and deals with the happy case
}